ui/hooks/useNetworkConnectionBanner.test.ts

```diff
@@ -0,0 +1,349 @@
+  return {
+    ...jest.requireActual('../store/actions'),
+    updateNetworkConnectionBanner: jest.fn(() => ({
+      type: 'UPDATE_NETWORK_CONNECTION_BANNER',
+    })),
+  };
+});
+
+jest.mock('../../shared/modules/selectors/networks', () => {
+  return {
+    ...jest.requireActual('../../shared/modules/selectors/networks'),
+    getNetworkConfigurationsByChainId: jest.fn(),
+  };
+});
+
+const mockSelectFirstUnavailableEvmNetwork = jest.mocked(
+  selectFirstUnavailableEvmNetwork,
+);
+const mockGetNetworkConnectionBanner = jest.mocked(getNetworkConnectionBanner);
+const mockUpdateNetworkConnectionBanner = jest.mocked(
+  updateNetworkConnectionBanner,
+);
+const mockGetNetworkConfigurationsByChainId = jest.mocked(
+  getNetworkConfigurationsByChainId,
+);
+
+describe('useNetworkConnectionBanner', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+
+    mockGetNetworkConfigurationsByChainId.mockReturnValue({
+      '0x1': {
+        name: 'Ethereum Mainnet',
+        chainId: '0x1',
+        nativeCurrency: 'ETH',
+        rpcEndpoints: [
+          {
+            networkClientId: 'mainnet',
+            url: 'https://mainnet.infura.io/v3/{infuraProjectId}',
+            type: RpcEndpointType.Infura,
+          },
+        ],
+        defaultRpcEndpointIndex: 0,
+        blockExplorerUrls: ['https://etherscan.io'],
+      },
+    });
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  describe('when all networks are available', () => {
+    it("updates the status of the banner to 'available' if not already updated", () => {
+      mockSelectFirstUnavailableEvmNetwork.mockReturnValue(null);
+      mockGetNetworkConnectionBanner.mockReturnValue({ status: 'unknown' });
+
+      renderHookWithProviderTyped(
+        () => useNetworkConnectionBanner(),
+        mockState,
+      );
+
+      expect(mockUpdateNetworkConnectionBanner).toHaveBeenCalledWith({
+        status: 'available',
+      });
+    });
+
+    it("does not update the status of the banner to 'available' if already updated", () => {
+      mockSelectFirstUnavailableEvmNetwork.mockReturnValue(null);
+      mockGetNetworkConnectionBanner.mockReturnValue({
+        status: 'available',
+        networkName: 'Ethereum Mainnet',
+        networkClientId: 'mainnet',
+        chainId: '0x1',
+      });
+
+      renderHookWithProviderTyped(
+        () => useNetworkConnectionBanner(),
+        mockState,
+      );
+
+      expect(mockUpdateNetworkConnectionBanner).not.toHaveBeenCalled();
+    });
+
+    it('does not create a MetaMetrics event', () => {
+      mockSelectFirstUnavailableEvmNetwork.mockReturnValue(null);
+      mockGetNetworkConnectionBanner.mockReturnValue({ status: 'unknown' });
+      const mockTrackEvent = jest.fn();
+
+      renderHookWithProviderTyped(
+        () => useNetworkConnectionBanner(),
+        mockState,
+        undefined,
+        undefined,
+        () => mockTrackEvent,
```

Comment:
This is how we create a mock version of `trackEvent`, pass it to the `render` helper, and test that it is not called. This is related to the changes to the helper we made.